django_mongodb/fields/json.py

```diff
@@ -0,0 +1,221 @@
+from django.db import NotSupportedError
+from django.db.models.fields.json import (
+    ContainedBy,
+    DataContains,
+    HasAnyKeys,
+    HasKey,
+    HasKeyLookup,
+    HasKeys,
+    JSONExact,
+    KeyTransform,
+    KeyTransformIn,
+    KeyTransformIsNull,
+    KeyTransformNumericLookupMixin,
+)
+
+from ..lookups import builtin_lookup
+from ..query_utils import process_lhs, process_rhs
+
+
+def contained_by(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contained_by lookup is not supported on this database backend.")
+
+
+def data_contains(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contains lookup is not supported on this database backend.")
+
+
+def _has_key_predicate(path, root_column, negated=False):
+    result = {"$and": [{"$ne": [{"$type": path}, "missing"]}, {"$ne": [root_column, None]}]}
+    if negated:
+        result = {"$not": result}
+    return result
+
+
+def has_key_lookup(self, compiler, connection):
+    """
+    Performs a key lookup in a JSONField to check for the existence of a key.
+
+    The key lookup in a JSONField is defined by the following conditions:
+    1. The path type must be different from "missing".
+    2. In cases where the JSONField is None (as the $type returns Null instead of "missing"),
+       an additional check for not null is added.
+
+    Returns:
+        A dictionary representing the MongoDB query for key lookup.
+
+    Raises:
+        AssertionError: If `self.mongo_operator` is None and there are multiple keys.
+
+    Note:
+        The function handles key lookup in two ways:
+        - Via `KeyTransform`
+        - Directly via key
+
+        Both are transformed into a `KeyTransform` for consistent handling.
+    """
+    rhs = self.rhs
+    lhs = process_lhs(self, compiler, connection)
+    if not isinstance(rhs, list | tuple):
+        rhs = [rhs]
+    paths = []
+    # Transform all keys into KeyTransform instances for consistent handling
+    for key in rhs:
+        rhs_json_path = key if isinstance(key, KeyTransform) else KeyTransform(key, self.lhs)
+        paths.append(rhs_json_path.as_mql(compiler, connection))
+    keys = []
+    for path in paths:
+        keys.append(_has_key_predicate(path, lhs))
+    if self.mongo_operator is None:
+        assert len(keys) == 1
```

Comment:
Was this mainly for debugging? Can we remove it now? We shouldn't rely on `assert` for anything critical since asserts are removed when running with `python -O`. If so, I'll make the update along with my next batch of language edits.

Reply:
Yes, will remove. It was for debugging. Shall we also remove this assert: https://github.com/mongodb-labs/django-mongodb/blob/main/django_mongodb/base.py#L121?

Reply:
I made a brief attempt to refactor the connection code to conform to Django's normal API, but it wasn't successful. I may return to it later. For now I wouldn't bother with a commit just to remove that line.